File: interface/src/main/java/cn/itcast/core/service/OrdersService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.entity.OrderEntity;
import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;
import cn.itcast.core.pojo.good.BrandQuery;
import cn.itcast.core.pojo.order.Order;
import org.apache.ibatis.annotations.Param;

import java.util.List;
import java.util.Map;

public interface OrdersService {


    public List<OrderEntity> findAll();
    public PageResult findPage(OrderEntity order, Integer page, Integer rows);

    public Brand findOne(Long id);


    public List<Map> selectOptionList();



}

File: web_shop/src/main/webapp/js/service/ordersService.js
// 定义服务层:
app.service("ordersService",function($http){
    this.findOrder = function(){
        return $http.get("../order/findOrder.do");
    }


    this.findPage=function(page,rows){
        return $http.get('../order/findPage.do?page='+page+'&rows='+rows);
    }


    this.findById=function(id){
        return $http.get("../order/findOne.do?id="+id);
    }

    this.search = function(page,rows,searchEntity){
        return $http.post("../order/findPage.do?page="+page+"&rows="+rows,searchEntity);
    }


    this.save = function(entity){
        return $http.post("../order/add.do",entity);
    }
});